.gitignore

```diff
@@ -41,3 +41,7 @@ config.rst
 
 package-lock.json
 geckodriver.log
+
+*.ipynb
+*.iml
+Part\ IA\ Computing\ (Michaelmas)
```

Comment:
I hope your course is going well, but we should keep the project's gitignore file clear of your personal files. ;-)

Also, we don't want to ignore `.ipynb` files in this repository, because we use some of them in docs and tests, e.g. https://github.com/jupyter/notebook/tree/master/docs/source/examples/Notebook

Not sure about .iml files.